cdb2api/cdb2api.c

```diff
@@ -5056,6 +5056,87 @@ int cdb2_bind_index(cdb2_hndl_tp *hndl, int index, int type,
     return 0;
 }
 
+/* bind array, varname is the string we used in the sql,
+ * count is the number of items in the array we will bind
+ * type is the type of elements we will bind ex. CDB2_INTEGER
+ * varaddr is the array address
+ * typelen is the type size of the acrual elements in the array (example sizeof(short))
+ * length optional param - is an array with lengths of each blob in the array
+ */
+int cdb2_bind_array(cdb2_hndl_tp *hndl, const char *varname, unsigned int count,
+                    int type, const void *varaddr[], int typelen, int length[])
+{
+    hndl->n_bindvars++;
+    hndl->bindvars = realloc(hndl->bindvars, sizeof(CDB2SQLQUERY__Bindvalue *) *
+                                                 hndl->n_bindvars);
+    CDB2SQLQUERY__Bindvalue *bindval = malloc(sizeof(CDB2SQLQUERY__Bindvalue));
+    cdb2__sqlquery__bindvalue__init(bindval);
+    bindval->type = type;
+    bindval->varname = (char *)varname;
+    switch(type) {
+    case CDB2_INTEGER:
+        if (typelen == sizeof(int32_t)) {
```

Comment:
We should probably also allow int16_t.  Or in the other direction, we should only allow int64, so they don't incur that extra conversion.

Reply:
I can even add int8_t...not sure what users would want. We can discuss.

Reply:
only int32 and int64 have native protobuf support on the Carray bind point of calling.